backend/src/org/commcare/session/SessionInstanceBuilder.java

```diff
@@ -39,7 +39,8 @@ private static void addSessionNavData(TreeElement sessionRoot, SessionFrame fram
      */
     private static void addDatums(TreeElement sessionData, SessionFrame frame) {
         for (StackFrameStep step : frame.getSteps()) {
-            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType())) {
+            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType()) ||
```

Comment:
Ahh gotcha. If you look at this line https://github.com/dimagi/commcare-core/blob/d98f844c7c95b17ec278d8ecc4cd8650845261a2/backend/src/org/commcare/session/CommCareSession.java#L563-L563

I changed the getFrame() call to perform a copy AND evaluate the unknowns to either commands or computeds, hence when they get to this point of the code they will have been evaluated to that type and need to be added.